src/components/ConnectWallet/ConnectedWalletChip.tsx

```diff
@@ -14,22 +20,44 @@ export default function ConnectedWalletChip({ disabled }: { disabled?: boolean }
   // TODO(kristiehuang): AccountDialog UI does not yet exist
   // const [open, setOpen] = useState(false)
 
+  // TODO: hover to see disconnect button is temporary; disconnection should live inside AccountDialog
+  const [hover, setHover] = useState(false)
+
   const { account } = useWeb3React()
+  function disconnectWallet() {
+    connections.forEach(([wallet, _]) => {
+      if (!(wallet instanceof Network || wallet instanceof Url)) {
+        // only deactivate non-network wallet connectors
+        wallet.deactivate ? wallet.deactivate() : wallet.resetState()
+      }
+    })
+  }
 
   return (
     <>
       <AccountButton
         hidden={disabled}
-        // onClick={() => console.log('open account modal')}
+        onClick={() => hover && disconnectWallet()}
```

Comment:
i'm not sure if you need hover to be positive here - i can't imagine how you would get into case where the state wouldn't update, but it seems like the safer thing is to disconnect on click always right?